src/azure-cli/azure/cli/command_modules/acr/cache.py

```diff
@@ -0,0 +1,108 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from ._utils import get_registry_by_name, get_resource_group_name_by_registry_name
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from azure.core.serialization import NULL as AzureCoreNull
+
+
+def acr_cache_show(cmd,
+                   client,
+                   registry_name,
+                   name):
+
+    rg = get_resource_group_name_by_registry_name(cmd.cli_ctx, registry_name, None)
+
+    return client.get(resource_group_name=rg,
+                      registry_name=registry_name,
+                      cache_rule_name=name)
+
+
+def acr_cache_list(cmd,
+                   client,
+                   registry_name):
+
+    rg = get_resource_group_name_by_registry_name(cmd.cli_ctx, registry_name, None)
+
+    return client.list(resource_group_name=rg,
+                       registry_name=registry_name)
+
+
+def acr_cache_delete(cmd,
+                     client,
+                     registry_name,
+                     name):
+
+    rg = get_resource_group_name_by_registry_name(cmd.cli_ctx, registry_name, None)
+
+    return client.begin_delete(resource_group_name=rg,
+                               registry_name=registry_name,
+                               cache_rule_name=name)
+
+
+def acr_cache_create(cmd,
+                     client,
+                     registry_name,
+                     name,
+                     source_repo,
+                     target_repo,
+                     cred_set=None):
+
+    registry, rg = get_registry_by_name(cmd.cli_ctx, registry_name, None)
+
+    cred_set_id = AzureCoreNull if not cred_set else f'{registry.id}/credentialSets/{cred_set}'
```

Comment:
what is the f'' for?

Reply:
It's just string interpolation (python calls them f-strings) to create the credential set resource id